app/utils.py

```diff
@@ -154,29 +162,141 @@ def setup_logger(name, level=logging.INFO):
     return logger
 
 
-def validate_resource(json):
-    validation_errors = {'errors': {'type': 'validation'}}
+def validate_resource(request, id=-1):
```

Comment:
This function has gotten pretty long and complex. I think we should break it down into parts and call all of those as separate functions. Like one function to check that there's no missing params, another function to validate the type, etc. Just something to break it down a bit.